lib/membrane_audio_mixer/adder.ex

```diff
@@ -11,26 +11,25 @@ defmodule Membrane.AudioMixer.Adder do
   alias Membrane.Caps.Audio.Raw
 
   @impl true
-  def init(), do: nil
+  def init(caps) do
+    size = Raw.sample_size(caps)
+    clipper = clipper_factory(caps)
 
-  @impl true
-  def mix(buffers, caps, state) do
-    sample_size = Raw.sample_size(caps)
+    %{caps: caps, clipper: clipper, sample_size: size}
```

Comment:
Why not struct? You could also get rid of the `mixer_module` from the element state as you can take the module from the struct with pattern matching:
```elixir
%module{} = %Mixer.State{}
assert module == Mixer.State
```